### PR TITLE
Feature/enqueue tests

### PR DESCRIPTION
**Pull request by iobrado**
This PR updates tests for Enqueue class with additional tests for methods from abstract classes. Currently only EnqueueThemeExampleTest.php contains attempts at testing methods from abstract classes.

### CODE REVIEW

File: tests/Enqueue/Theme/EnqueueThemeExampleTest.php [hunk 1]
```diff
@@ -13,6 +14,17 @@
...
+	mock('alias:EightshiftBoilerplate\Config\Config')
+	->shouldReceive('getProjectName', 'getProjectVersion')
+	->andReturn('tests/data', '1.0');
+	
+	// Functions\when('wp_register_style')->justReturn();
+	// Functions\when('wp_enqueue_style')->justReturn();
+	
+	// Functions\when('wp_register_script')->justReturn();
+	// Functions\when('wp_enqueue_script')->justReturn();
+	// Functions\when('wp_localize_script')->justReturn(true);
```
Comment (dingo-d) [inline tests/Enqueue/Theme/EnqueueThemeExampleTest.php:26]: This should be removed

File: tests/Enqueue/Theme/EnqueueThemeExampleTest.php [hunk 2]
```diff
@@ -30,3 +42,51 @@
...
+	$assetsPrefix = $this->example->getAssetsPrefix();
+
+	$this->assertIsString($assetsPrefix, 'getAssetsPrefix method must return a string');
+});
+
+test('getAssetsVersion method will return string', function () {
+	$assetsVersion = $this->example->getAssetsVersion();
+
+	$this->assertIsString($assetsVersion, 'getAssetsVersion method must return a string');
+});
```
Comment (dingo-d) [inline tests/Enqueue/Theme/EnqueueThemeExampleTest.php:56]: This is covered in the example test, no?

File: tests/Enqueue/Theme/EnqueueThemeExampleTest.php [hunk 3]
```diff
@@ -30,3 +42,51 @@
...
+
+test('enqueueStyles method will enqueue styles in a theme', function () {
+	Functions\when('wp_register_style')->alias(function($args) {
+		putenv("REGISTER_STYLE={$args}");
+	});
+
+	Functions\when('wp_enqueue_style')->alias(function($args) {
+		putenv("ENQUEUE_STYLE={$args}");
```
Comment (dingo-d) [inline tests/Enqueue/Theme/EnqueueThemeExampleTest.php:64]: Make sure you do a cleanup of these env variables. You can add `afterEach` and just place `putenv('...=');` there.

File: tests/Enqueue/Theme/EnqueueThemeExampleTest.php (hunk 2 above)
Comment (iobrado) [reply tests/Enqueue/Theme/EnqueueThemeExampleTest.php:56]: It's possible. This was one of the earlier tests I've written so it's possible I didn't even know how to check coverage then 😅

File: tests/Enqueue/Theme/EnqueueThemeExampleTest.php [hunk 4]
```diff
@@ -13,6 +14,17 @@
 	Monkey\setUp();
 	setupMocks();
 
+	mock('alias:EightshiftBoilerplate\Config\Config')
+	->shouldReceive('getProjectName', 'getProjectVersion')
+	->andReturn('tests/data', '1.0');
```
Comment (dingo-d) [inline tests/Enqueue/Theme/EnqueueThemeExampleTest.php:19]: Can you indent this?

**Comment by iruzevic** [comment]
@iobrado is this PR fixed?

**Comment by iobrado** [comment]
I've updated the EnqueueThemeExampleTest.php according to @dingo-d's instructions. If everything is fine, I'll update other tests accordingly. Hopefully will be done today.

File: tests/Enqueue/Admin/EnqueueAdminExampleTest.php [hunk 5]
```diff
@@ -3,31 +3,105 @@
...
-	$this->example = new EnqueueAdminExample($manifest);
+	$this->example = new EnqueueAdminTest($manifest);
+
+	$this->hookSuffix = 'test';
+
 });
 
 afterEach(function() {
 	Monkey\tearDown();
+	putenv('...=');
```
Comment (dingo-d) [inline tests/Enqueue/Admin/EnqueueAdminExampleTest.php:70]:
```suggestion
	putenv("REGISTER_STYLE");
	putenv("ENQUEUE_STYLE");
	putenv("REGISTER_SCRIPT");
	putenv("ENQUEUE_SCRIPT");
```

File: tests/Enqueue/Blocks/EnqueueBlockExampleTest.php [hunk 6]
```diff
@@ -43,16 +59,36 @@
...
-	$this->eb = new EnqueueBlocksExample($manifest);
+	$this->eb = new EnqueueBlocksTest($manifest);
 });
 
 /**
  * Cleanup after each test.
  */
 afterEach(function() {
 	Monkey\tearDown();
+	putenv('...=');
```
Comment (dingo-d) [inline tests/Enqueue/Blocks/EnqueueBlockExampleTest.php:91]: ☝🏼

File: tests/Enqueue/Theme/EnqueueThemeExampleTest.php [hunk 7]
```diff
@@ -3,30 +3,97 @@
...
+	Functions\when('wp_localize_script')->justReturn(putenv("SIDEAFFECT={$localize}"));
+
 	$manifest = new ManifestExample();
-	$this->example = new EnqueueThemeExample($manifest);
+	$this->example = new EnqueueThemeTest($manifest);
 });
 
 afterEach(function() {
 	Monkey\tearDown();
+	putenv('...=');
```
Comment (dingo-d) [inline tests/Enqueue/Theme/EnqueueThemeExampleTest.php:62]: ☝🏼

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left small comments for cleanup of env vars 🙂

**Comment by iobrado** [comment]
I've updated the code with better cleanup of env vars, so if everything else is fine, I believe this can be merged. There's only one small issue which causes the coverage for Enqueue classes to be 99.29%.

When registering `EnqueueAdminTest` class for adding some values to `getLocalizations()` protected method, there's no coverage for that method in tests. But when I remove the test class and let it run with `EnqueueAdminExample` class, there is coverage for `getLocalizations()` method, but then `wp_localize_script` in `AbstractEnqueueAdmin.php` isn't covered.

I really have no idea what's causing this, but other than that, everything else is covered just fine.

**Review by dingo-d — APPROVED** [review]
All works now. Great job 👍🏼

**Review by iruzevic — APPROVED** [review]

**Pull request merged**